internal/certificate/cert.go

```diff
@@ -0,0 +1,202 @@
+	x509.KeyUsageEncipherOnly:      "Encipher Only",
+	x509.KeyUsageDecipherOnly:      "Decipher Only",
+}
+
+// ExtKeyUsagesNameMap is a map of x509.Certificate ExtKeyUsages map used
+// in 'notation cert show' command
+var ExtKeyUsagesNameMap = map[x509.ExtKeyUsage]string{
+	x509.ExtKeyUsageAny:                            "Any",
+	x509.ExtKeyUsageServerAuth:                     "TLS Web Server Authentication",
+	x509.ExtKeyUsageClientAuth:                     "TLS Web Client Authentication",
+	x509.ExtKeyUsageCodeSigning:                    "Code Signing",
+	x509.ExtKeyUsageEmailProtection:                "E-mail Protection",
+	x509.ExtKeyUsageIPSECEndSystem:                 "IPSec End System",
+	x509.ExtKeyUsageIPSECTunnel:                    "IPSec Tunnel",
+	x509.ExtKeyUsageIPSECUser:                      "IPSec User",
+	x509.ExtKeyUsageTimeStamping:                   "Time Stamping",
+	x509.ExtKeyUsageOCSPSigning:                    "OCSP Signing",
+	x509.ExtKeyUsageMicrosoftServerGatedCrypto:     "Microsoft Server Gated Crypto",
+	x509.ExtKeyUsageNetscapeServerGatedCrypto:      "Netscape Server Gated Crypto",
+	x509.ExtKeyUsageMicrosoftCommercialCodeSigning: "Microsoft Commercial Code Signing",
+	x509.ExtKeyUsageMicrosoftKernelCodeSigning:     "Microsoft Kernel Code Signing",
+}
+
+// AddCertCore adds a single cert file at path to the trust store
+// under dir truststore/x509/storeType/namedStore
+func AddCertCore(path, storeType, namedStore string, display bool) error {
+	// initialize
+	certPath, err := filepath.Abs(path)
+	if err != nil {
+		return err
+	}
+	storeType = strings.TrimSpace(storeType)
+	if storeType == "" {
+		return errors.New("store type cannot be empty or contain only whitespaces")
+	}
+	namedStore = strings.TrimSpace(namedStore)
+	if namedStore == "" {
+		return errors.New("named store cannot be empty or contain only whitespaces")
+	}
+
+	// check if the target path is a cert (support PEM and DER formats)
+	if _, err := corex509.ReadCertificateFile(certPath); err != nil {
+		return err
+	}
+
+	// core process
+	// get the trust store path
+	trustStorePath, err := dir.Path.UserConfigFS.GetPath(dir.TrustStoreDir, "x509", storeType, namedStore)
+	if err := CheckError(err); err != nil {
+		return err
+	}
+	// check if certificate already in the trust store
+	if _, err := os.Stat(filepath.Join(trustStorePath, filepath.Base(certPath))); err == nil {
+		return errors.New("certificate already exists in the Trust Store")
+	}
+	// add cert to trust store
+	_, err = osutil.Copy(certPath, trustStorePath)
+	if err != nil {
+		return err
+	}
+
+	// write out
+	if display {
+		fmt.Printf("Successfully added %s to named store %s of type %s\n", filepath.Base(certPath), namedStore, storeType)
+	}
+
+	return nil
+}
+
+// ListCertsCore walks through root and lists all x509 certificates in it
+func ListCertsCore(root string) error {
+	return filepath.WalkDir(root, func(path string, d fs.DirEntry, err error) error {
```

Comment:
Can we use https://github.com/notaryproject/notation-go/blob/f35c52efb5cc24ba8dc6371b95f69098ff403987/verification/store.go#L23 instead?

That function also performs all the validations on the trust store

Reply:
These two functions have fundamental differences. [LoadX509TrustStore](https://github.com/notaryproject/notation-go/blob/f35c52efb5cc24ba8dc6371b95f69098ff403987/verification/store.go#L48) will returns an error if joinedPath is not a regular file. However, in `notation cert list`, we won't show an eror in this situation. Instead, it would only list regular files, if there's no such file, `notation cert list` won't print anything in the command line.